bundles/org.openhab.binding.mideaac/src/main/java/org/openhab/binding/mideaac/internal/MideaACConfiguration.java

```diff
@@ -0,0 +1,270 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.mideaac.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+
+/**
+ * The {@link MideaACConfiguration} class contains fields mapping thing configuration parameters.
+ *
+ * @author Jacek Dobrowolski - Initial contribution
+ * @author Bob Eckhoff - JavaDoc
+ */
+@NonNullByDefault
+public class MideaACConfiguration {
+
+    private String ipAddress = "";
```

Comment:
You could remove all the getters and setters and just have the members pulbic.
While you declare them you could set the default value.

Example:
```
    public String ipAddress = "";
    public String ipPort = 6444;
```

I also noticed the port is a String, while it can only be numeric, so why not make it an int?

Reply:
Good idea to make the variables public.  Reduced a lot of lines of code and easier to understand. As to the int ipPort versus String ipPort a lot of changes would be required and only save one line of code when the variable is used in creating the socket. int port = Integer.parseInt(ipPort);

Reply:
I'm still hung up a bit with this change. It is related to my other question about how the discovered properties become configurations.  Although there is a default value, the ipPort is discovered in the DiscoveryService converted to a string, 
```
byte[] portIdBytes = Utils.reverse(Arrays.copyOfRange(reply, 4, 8));
            BigInteger portId = new BigInteger(1, portIdBytes);
            mSmartPort = portId.toString(10);
            logger.debug("Port: '{}'", mSmartPort);
```
then put into properties, but ends up in the configuration.  Also in the binding constants everything is a String
My unease was just not to fix something that isn't broke.

Reply:
Changing the mSmartPort to int and adjusting this line should do the trick:
`mSmartPort = portId.toString(10);`
to
`mSmartPort = portId.intValue();`

Reply:
While thinking about this there is a potential issue. I think this is the sequence. The `DiscoveryService` (from inBox scan) saves the `ipPort`, `version`, `deviceId` and `ipAddress `as properties, but when the initialize() `(MideaACHandler)` is first run it will pick up the default configuration from `MideaACConfiguration`, realize they are not valid, run DiscoveryService again, but this time cache them as DiscoveryResult, then assign them to the configuration (at least the `ipPort `and `deviceId`).  this would need to be changed. 
```
        Object propertyIpPort = Objects.requireNonNull(discoveryProps.get(CONFIG_IP_PORT));
        configuration.put(CONFIG_IP_PORT, propertyIpPort.toString());
```
 I need to think a bit on this.  Historically you could go to the python app and get the deviceId and ipPort, but maybe these should just be properties